src/main/java/teammates/common/datatransfer/questions/FeedbackContributionQuestionDetails.java

```diff
@@ -44,21 +44,35 @@ public class FeedbackContributionQuestionDetails extends FeedbackQuestionDetails
 
     private static final Logger log = Logger.getLogger();
 
+    private boolean isZeroSum;
     private boolean isNotSureAllowed;
 
     public FeedbackContributionQuestionDetails() {
-        this(null);
+        super(FeedbackQuestionType.CONTRIB, null);
+        /*
+            Contribution question details was changed to include isZeroSum field in
+            https://github.com/TEAMMATES/teammates/pull/11827.
+            isZeroSum field has to be set to false for old contribution questions.
+            This constructor is used to deserialize question details string into FeedbackContributionQuestionDetails
+            object.
+            We set the default value of isZeroSum to false here so that the default value if isZeroSum is not present
+            will be false.
+        */
+        isZeroSum = false;
+        isNotSureAllowed = false;
     }
 
     public FeedbackContributionQuestionDetails(String questionText) {
         super(FeedbackQuestionType.CONTRIB, questionText);
-        isNotSureAllowed = true;
+        isZeroSum = true;
```

Comment:
Why is it `true` here but `false` up there?

Reply:
The difference in values for `isZeroSum` is to achieve backward compatibility. The default constructor, i.e. `FeedbackContributionQuestionDetails()`, is used to deserialize question details. For old questions, since `isZeroSum` is absent, the value for `isZeroSum` will be set to `false` by default; For new questions, since `isZeroSum` is present, the corresponding value will overwrite the default value.

Reply:
I can't say I like the way it's done. Will there be a plan to migrate data such that `isZeroSum` exists everywhere?

Reply:
Sure, if we're good with data migration, it simplifies the logic too.